src/generic_solver.cpp

```diff
@@ -478,31 +485,81 @@ Sort GenericSolver::make_sort(SortKind sk, const SortVec & sorts) const
 
 Sort GenericSolver::make_sort(const DatatypeDecl & d) const
 {
-  throw NotImplementedException("Generic Solvers do not support datatypes");
+  std::string dtName = (*datatypedecl_name_map)[d];
+  if (name_sort_map->find(dtName) == name_sort_map->end()) {
+    std::string toSolver = "(" + DECLARE_DATATYPE_STR + " ((";
```

Comment:
Let's keep variable names with lower case and underscores. Same for other variable names.